convention-gradle-enterprise-gradle-plugin/plugins/gradle-5-or-newer/src/main/java/com/myorg/ConventionGradleEnterpriseGradlePlugin.java

```diff
@@ -44,21 +45,28 @@ public void apply(Object target) {
     private void configureGradle6OrNewer(Settings settings) {
         settings.getPluginManager().apply(GradleEnterprisePlugin.class);
         settings.getPluginManager().apply(CommonCustomUserDataGradlePlugin.class);
-        configureGradleEnterprise(settings.getExtensions().getByType(GradleEnterpriseExtension.class));
+        configureGradleEnterprise(settings.getExtensions().getByType(GradleEnterpriseExtension.class), settings.getGradle().getStartParameter());
         configureBuildCache(settings.getBuildCache(), settings.getExtensions().getByType(GradleEnterpriseExtension.class));
     }
 
     private void configureGradle5(Project project) {
         project.getPluginManager().apply(BuildScanPlugin.class);
         project.getPluginManager().apply(CommonCustomUserDataGradlePlugin.class);
-        configureGradleEnterprise(project.getExtensions().getByType(GradleEnterpriseExtension.class));
+        configureGradleEnterprise(project.getExtensions().getByType(GradleEnterpriseExtension.class), project.getGradle().getStartParameter());
         // configureBuildCache is not called because the build cache cannot be configured via a plugin prior to Gradle 6.0
     }
 
-    private void configureGradleEnterprise(GradleEnterpriseExtension gradleEnterprise) {
+    private void configureGradleEnterprise(GradleEnterpriseExtension gradleEnterprise, StartParameter startParameter) {
         // CHANGE ME: Apply your Gradle Enterprise configuration here
         gradleEnterprise.setServer("https://enterprise-samples.gradle.com");
-        configureBuildScan(gradleEnterprise.getBuildScan());
+        if (!startParameter.isNoBuildScan() && !containsPropertiesTask(startParameter)) {
```

Comment:
Is the `!startParameter.isNoBuildScan()` needed here? It seems the check for `properties` tasks would be sufficient to accomplish the goal here.

Reply:
Good point. The logic was taken from here:

https://github.com/gradle/gradle-org-conventions-plugin/blame/fc264a151c479c44ec057d4baef39c6844788619/src/main/java/com/gradle/enterprise/conventions/GradleEnterpriseConventionsPlugin.java#L55

Reply:
Good point. The logic was taken from here:

https://github.com/gradle/gradle-org-conventions-plugin/blame/fc264a151c479c44ec057d4baef39c6844788619/src/main/java/com/gradle/enterprise/conventions/GradleEnterpriseConventionsPlugin.java#L55

But it is definitely redundant. I will remove it.

Reply:
Updated in 4c0994db4d8576f49d2d165617eba8e1f71e6d76